src/Ser/SerCsv.jl

```diff
@@ -3,8 +3,77 @@ module SerCsv
 export to_csv
 
 using Dates
+
 import ..to_flatten
 
+issimple(::Any)::Bool = false
+issimple(::AbstractString)::Bool = true
+issimple(::Symbol)::Bool = true
+issimple(::AbstractChar)::Bool = true
+issimple(::Number)::Bool = true
+issimple(::Enum)::Bool = true
+issimple(::Dates.TimeType)::Bool = true
+
+issimple(t::Type) = (
+    t <: AbstractString || t <: Symbol || t <: AbstractChar || t <: Number || t <: Enum || t <: Type || t <: Dates.TimeType)
+
+isnull(t::Type) = (t === Nothing || t === Missing)
+isnull(t::Any) = isnothing(t) || ismissing(t)
+
+isunion(t::Type) = t isa Union
+
+iscomposite(t::Type) = !issimple(t) && !isnull(t) && !isunion(t)
+
+function is_valid_union(t::Type)::Bool
+    if !isunion(t)
+        return throw("Parameter `t` must be a Union type")
+    end
+    simple_count = 0
+    composite_count = 0
+    for type in Base.uniontypes(t)
+        if issimple(type)
+            simple_count += 1
+        elseif iscomposite(type)
+            composite_count += 1
+        end
+    end
+    return simple_count + composite_count == 1
+end
+
+"""
+Extract first composite type from union type, or return itself is `t` is a Union type, return nothing if no composite type.
```

Comment:
Typo:  Extract first composite type from union type, or return itself if `t` is a composite type, return nothing if no composite type.